src/pkg/distribution/distribution.go

```diff
@@ -47,8 +47,12 @@ var (
 var (
 	name      = fmt.Sprintf("(?P<name>%s)", ref.NameRegexp)
 	reference = fmt.Sprintf("(?P<reference>((%s)|(%s)))", ref.DigestRegexp, ref.TagRegexp)
+	dgt       = fmt.Sprintf("(?P<digest>%s)", ref.DigestRegexp)
 	sessionID = "(?P<session_id>[a-zA-Z0-9-_.=]+)"
 
+	// BlobURLRegexp regexp which match blob url
+	BlobURLRegexp = regexp.MustCompile(`^/v2/` + name + `/blobs/` + dgt)
```

Comment:
duplicate with `V2BlobUploadURLRe` in ./ser/server/middlware/patterns.go?

Reply:
I think parse the pattern paramters from /v2 url need to put in this pkg not in src/server/middleware/patterns.go

I will create a issue to track this problem.